views/components/Graph/index.jsx

```diff
@@ -109,21 +110,19 @@ export const Graph = ({ study, subject, user, theme, setNotification }) => {
     }
 
     graphRef.current = new Matrix(graphRef.current, matrixProps)
-
     graphRef.current.create(graph.matrixData)
+
+    setShouldDrawPng(true)
```

Comment:
I wonder if we want to store `graphRef.current.lastChild` in state instead of a boolean. That way we can remove a conditional in the useEffect

```suggestion
    setGraphNode(graphRef.current.lastChild)
```

```js
React.useEffect(() => {
  if (graphNode) {
    const svgString = new XMLSerializer().serializeToString(graphNode)
    const kanvas = canvasRef.current
    kanvas.width = graphNode.getBBox().width
    kanvas.height = graphNode.getBBox().height
    const svgUrl =
      'data:image/svg+xml; charset=utf8, ' +
      encodeURIComponent(
        svgString.replace(
          '<svg xmlns="http://www.w3.org/2000/svg" width="100%" height="100%">',
          `<svg xmlns="http://www.w3.org/2000/svg" width="${kanvas.width}" height="${kanvas.height}">`
        )
      )
    // png conversion
    const img = new Image(kanvas.width, kanvas.height)
    const ctx = kanvas.getContext('2d')
    img.src = svgUrl
    img.onload = () => {
      ctx.drawImage(img, 0, 0)
    }
  }
}, [graphNode])
```

Reply:
Thanks for the suggestion, yesterday this was originally how I wanted to implement and I learned that at this point of rendering the graph, the last child is still undefined. That's why I had to go with a boolean.

<img width="1439" alt="Screenshot 2024-04-18 at 8 28 32 AM" src="https://github.com/AMP-SCZ/dpdash/assets/19805355/5ecd1bed-e323-4ea9-be4f-d02bea065de7">